apiserver/apiserver.go

```diff
@@ -252,12 +264,46 @@ func (c RateLimitConfig) Validate() error {
 	return nil
 }
 
+// LogSinkConfig holds parameters to control the API server's
+// logsink endpoint behaviour.
+type LogSinkConfig struct {
+	// DBLoggerBufferSize is the capacity of the database logger's buffer.
+	DBLoggerBufferSize int
+
+	// DBLoggerFlushInterval is the amount of time to allow a log record
+	// to sit in the buffer before being flushed to the database.
+	DBLoggerFlushInterval time.Duration
+}
+
+// Validate validates the logsink endpoint configuration.
+func (cfg LogSinkConfig) Validate() error {
+	if cfg.DBLoggerBufferSize <= 0 || cfg.DBLoggerBufferSize > 1000 {
+		return errors.NotValidf("DBLoggerBufferSize %d <= 0 or > 1000", cfg.DBLoggerBufferSize)
+	}
+	if cfg.DBLoggerFlushInterval <= 0 {
```

Comment:
Can we put an upper bound on this. Like we did for conn throttling config. We want to prevent the user from specifying something stupid.